go/libraries/doltcore/sqle/tables.go

```diff
@@ -183,8 +183,8 @@ var _ doltReadOnlyTableInterface = (*DoltTable)(nil)
 //var _ sql.ProjectedTable = (*DoltTable)(nil)
 
 // IndexedAccess implements sql.IndexAddressableTable
-func (t *DoltTable) IndexedAccess(idx sql.Index) sql.IndexedTable {
-	return NewIndexedDoltTable(t, idx.(index.DoltIndex))
+func (t *DoltTable) IndexedAccess(lookup sql.IndexLookup) sql.IndexedTable {
```

Comment:
This seems bad... were we not getting indexes before this change? How did this even work?

Reply:
the index was passed but not the range. so commit indexes are the first that don't support every range, or we didn't have enough tests to identify when other tables should be rejecting index lookups